starter_code/src/movies.js

```diff
@@ -1,15 +1,108 @@
 /* eslint no-restricted-globals: 'off' */
-
 // Iteration 1: Ordering by year - Order by year, ascending (in growing order)
 
-// Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct
+let orderByYear = function(array){
+    //Creando una copia del arreglo original (movies)
+    let moviesYear = array.map(function (element){
+        return element;
+    });
+
+   moviesYear.sort(function(a,b){
+    if(a.year> b.year) return 1;
+    if (a.year< b.year) return -1;
+    if(a.title >b.title ) return  1;
+
+    });
+  
+    return moviesYear;
+
+};
 
+// Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct
+let howManyMovies = function(array){
+    let arrayHowMany =array.filter(element => {
+       return element.director === "Steven Spielberg" && element.genre.includes("Drama");
+     
+      });
+    
+     return arrayHowMany.length;
+ }
 // Iteration 3: Alphabetic Order - Order by title and print the first 20 titles
+let orderAlphabetically = function (array){
+    let firstElements=[];
+
+     if(array.length < 20){
+       
+        firstElements = array.map(function(element){
+       
+          return element.title;
+       });
+       return firstElements.sort();
+       
+     }else if(array.length > 20){
+           
+        firstElements =array.slice(0,20);
+        return firstElements.sort();
 
+    }
+  
+};
+orderAlphabetically(movies);
 // Iteration 4: All rates average - Get the average of all rates with 2 decimals
+let ratesAverage = function(array){
```

Comment:
this solution can be improve, like a first mapping the movies, to get one new array with only rates, after that use `reduce` to get the sum and get the average.